showcase/src/app/app.component.ts

```diff
@@ -68,6 +70,21 @@ export class AppComponent implements OnInit {
 		this.breakpointObserver.observe([this.mediaQueryMdSm]).subscribe((state: BreakpointState) => {
 			this.isMenuModeActive = state.matches;
 		});
+
+		const scrollToHash = (tran: Transition): void => {
+			const hash = tran.targetState().params()["#"];
+
+			// Document not loaded on first run yet
+			setTimeout(() => {
+				const element: HTMLElement | null = document.querySelector(`#${hash}`);
+				if (element) {
+					element.scrollIntoView();
+				}
+			});
+		};
+
+		this.routingService.addTransitionHook(StarkRoutingTransitionHook.ON_FINISH, {}, scrollToHash);
+		this.routingService.addTransitionHook(StarkRoutingTransitionHook.ON_ERROR, {}, scrollToHash);
```

Comment:
I don't think you need the `ON_ERROR` hook here. I know that UI Router calls this hook whenever a navigation is triggered to the same state but I would say that this should be fixed in the `StarkRoutingService` or maybe in the UI Router itself if applicable, so you can simply remove this line ;)